File: 24-anagram-checker/AnagramChecker.java
import java.util.Arrays;

class AnagramChecker {
  public static void main(String[] args) {
    System.out.println("Enter two strings and I'll tell you if they are anagrams:");
    var first = System.console().readLine("Enter the first string: ");
    var second = System.console().readLine("Enter the second string: ");

    var isAnagram = isAnagram(first, second);

    System.out.printf("\"%s\" and \"%s\" are %s%n", first, second, (isAnagram ? "anagrams" : "not anagrams"));
  }

  private static boolean isAnagram(String first, String second) {
    if (first != null && second != null && first.length() > 0 && first.length() == second.length()) {
      var firstChars = first.toCharArray();
      var secondChars = second.toCharArray();
      Arrays.sort(firstChars);
      Arrays.sort(secondChars);
      return Arrays.equals(firstChars, secondChars);
    }

    return false;
  }
}

File: 24-anagram-checker/anagram_checker.py
def is_anagram(first, second):
    if first and second and len(first) > 0 and len(first) == len(second):
        if sorted(first.lower()) == sorted(second.lower()):
            return True
    return False

print("Enter two strings and I'll tell you if they are anagrams:")
first = input("Enter the first string: ")
second = input("Enter the second string: ")
is_anagram = is_anagram(first, second)
print(f'"{first}" and "{second}" are ' \
      f'{"anargrams" if is_anagram else "not anagrams"}.')

